### ISSUE TITLE
Code gives better results than that mentioned in the paper

### ISSUE THREAD

**Issue by sampathchanda** [wzzheng/HDML#6]
Hi @wzzheng ,

Thanks for your interesting paper and open sourcing the code.

I tried retraining with the default parameters in the FLAGS.py and with Cars196 dataset. Following are the final evaluation results that I see:
```
num_correct: 6509 num: 8131 K: 1, Recall: 0.801
num_correct: 7154 num: 8131 K: 2, Recall: 0.880
num_correct: 7576 num: 8131 K: 4, Recall: 0.932
num_correct: 7806 num: 8131 K: 8, Recall: 0.960
num_correct: 7948 num: 8131 K: 16, Recall: 0.977
num_correct: 8030 num: 8131 K: 32, Recall: 0.988
NMI: 0.7029318395368825
F1: 0.4271278961451372
```

These are around 1% higher than the numbers mentioned in the paper. Any particular reason why this is the case?

Also, with the default options, but just changing the dataset to CUB, I see the training going to NaN immediately. Are there any hyper parameters that are to be changed for training with the CUB dataset?

**Comment by trungpham2606** [comment]
@sampathchanda Hi, Iam also trying this repository, but Iam having problems with Hdf5, it seems that with HDF5 we cant shuffle the order of index in it. How can you re-train this repo ? 

This is the error i got

![image](https://user-images.githubusercontent.com/35657426/58929212-2a7e6180-8780-11e9-8bc0-bb41bfde89b0.png)

**Comment by trungpham2606** [comment]
@sampathchanda  Hello, Iam also trying this repository. Can you tell me how many epochs you needed to achieve that results :-? Btw, have you tried to test this model with unknown data to verify whether or not this model can distinguish 2 images (at the same class or not) ?

**Comment by sampathchanda** [comment]
@trungpham2606 I got the above results in 20K iterations, which is part of the default configuration. No, I haven't tried with unknown data.

**Comment by trungpham2606** [comment]
@sampathchanda  Can you please send me your pre-trained weights ?, Iam working in distinguish images on my own dataset.

**Comment by wzzheng** [comment]
@sampathchanda Very sorry for the late reply. We did achieve better results than those in the paper since we only reported average numbers. So your results are not surprising :) Btw, the best result we got is about 0.81 of recall@1. Also, we changed the initialization method from tf.truncated_normal_initializer(stddev=0.1) to tf.glorot_uniform_initializer() since the latter leads to more stable performance. 
For CUB dataset, there should be no problem. I will include hyperparameters for CUB soon, hopefully before sunset.

**Comment by trungpham2606** [comment]
@wzzheng When can you public your pre-trained weights :3 Iam so excited looking forward to it

**Comment by wzzheng** [comment]
@trungpham2606 Hi, if you could tell me your email address, I will send a pretrained model to you.

**Comment by wzzheng** [comment]
@sampathchanda Hi, I've added the recommended hyperparameters for CUB-200-2011, I hope they will solve your problem.

**Comment by trungpham2606** [comment]
@wzzheng here's my gmail: trungpham2606@gmail.com Thank you so much! Iam also using your model to solve images classification problem by image similarity. But the results I got so far are not good enough, maybe due to the lack of training iterations.
